src/Lobby.tsx

```diff
@@ -0,0 +1,78 @@
+import React from "react";
+import "./App.css";
+import GameHUD from "./GameHUD";
+import { checkLogin } from "./Login";
+import { useParams } from "react-router-dom";
+import io from "socket.io-client";
+
+interface ILobby {}
+
+const Lobby = (props: ILobby) => {
+  const [loggedIn, setLoggedIn] = React.useState<boolean>(false);
+  const [users, setUsers] = React.useState<string[]>([]);
+  const [email, setEmail] = React.useState<string>("");
+  const [socket, setSocket] = React.useState(() => io("/"));
+  const [update, setUpdate] = React.useState<string>("");
+  const { id } = useParams();
+
+  React.useEffect(() => {
+    const asyncOps = async () => {
+      const loginStatus = await checkLogin();
```

Comment:
The name `loginStatus` leads me to believe it's an enum with supported values like `LOGGED_IN` or `LOGGED_OUT`, but it turns out to be an object with all kinds of properties. It'd be less jarring and easier to reason about if `loginStatus` were renamed `currentUser`, and the `checkLogin` function were renamed `getCurrentUser`. Then the lines below can be refactored into stuff that's more readable, like `socket.emit("joinLobby", lobbyId, currentUser.email)`